### PR TITLE
New filters for events ( depended on backend)

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

**Comment by wissog** [comment]
image
the spaces from the right side of the filter (when your language is HEB) and above it - can you get rid of them?
the functionality seems fine

File: src/sagas/lists.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { call, put, select, takeEvery, takeLatest } from 'redux-saga/effects';
 import { filtersTransformer } from '../filters';
 import Api from '../helpers/Api';
-import { COLLECTION_LESSONS_TYPE, CT_VIDEO_PROGRAM_CHAPTER, UNIT_LESSONS_TYPE } from '../helpers/consts';
```
Comment (bbfsdev) [inline src/sagas/lists.js:4]: Why you remove UNIT_LESSONS_TYPE?

File: src/sagas/lists.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ import { getQuery, pushQuery } from './helpers/url';
 import { fetchCollectionsByIDs, fetchUnitsByIDs } from './mdb';
 import { fetchViewsByUIDs } from './recommended';
 
+const endpointByNamespace = {
+  [PAGE_NS_LESSONS]: Api.lessons,
+  [PAGE_NS_EVENTS]: Api.events
```
Comment (bbfsdev) [inline src/sagas/lists.js:17]: You area adding events to fetch by redux namespace?

File: src/sagas/lists.js [hunk 3]
```diff
@@ -47,45 +52,33 @@ function* fetchList(action) {
   }
 }
 
-function* fetchListLessons(action) {
-  const { withViews = false, namespace, ...args } = action.payload;
```
Comment (bbfsdev) [inline src/sagas/lists.js:51]: withViews is not relevant? You never use it?

File: src/sagas/lists.js [hunk 4]
```diff
@@ -47,45 +52,33 @@ function* fetchList(action) {
...
+function* fetchSectionList(action) {
+  const { namespace, ...args } = action.payload;
 
   const filters      = yield select(state => filterSelectors.getFilters(state.filters, namespace));
   const filterParams = filtersTransformer.toApiParams(filters) || {};
 
   const language = yield select(state => settings.getLanguage(state.settings));
 
   try {
-    const { data: { items, content_units, total } } = yield call(Api.lessons, { ...args, ...filterParams, language });
```
Comment (bbfsdev) [inline src/sagas/lists.js:59]: Why do you NOT need content_units / total variables any more?

File: src/sagas/lists.js [hunk 5]
```diff
@@ -47,45 +52,33 @@ function* fetchList(action) {
...
+    const { data } = yield call(endpointByNamespace[namespace], { ...args, ...filterParams, language });
 
-    let cuIDs;
-    let cuIDsView;
-    let cIDs;
-    const data = { total, items };
+    const { items } = data;
 
-    if (content_units) {
```
Comment (bbfsdev) [inline src/sagas/lists.js:66]: What is the case of 
```if (content_units) {```
That you don't need any more?

File: src/sagas/filtersAside.js [hunk 6]
```diff
@@ -18,7 +18,8 @@ const defaultStatParams    = {
   with_content_types: true,
   with_persons: false,
   with_original_languages: false,
-  with_media: false
+  with_media: false,
```
Comment (bbfsdev) [inline src/sagas/filtersAside.js:21]: What are those params for?

File: src/sagas/filtersAside.js [hunk 7]
```diff
@@ -62,14 +63,14 @@ export function* fetchStat(action) {
 
     const responses = yield all(requests);
 
-    const { data: dataCU } = responses.shift();
-    const dataC            = countC ? responses.shift()?.data : {};
-    const dataL            = countL ? responses.shift()?.data : {};
+    const { data: dataCU }                  = responses.shift();
+    const { data: { locations, ...dataC } } = countC ? responses.shift() : { data: false };
```
Comment (bbfsdev) [inline src/sagas/filtersAside.js:67]: what if responses.shift().data is not defined?

File: src/redux/modules/filtersAside.js [hunk 8]
```diff
@@ -137,18 +173,21 @@ export const reducer = handleActions({
...
 /* Selectors */
-const getStats  = (state, ns, fn) => id => state[ns]?.[fn]?.byId[id] || 0;
-const getTree   = (state, ns, fn) => state[ns]?.[fn]?.tree || [];
-const isReady   = (state, ns) => !!state[ns]?.isReady;
-const getWipErr = (state, ns) => ({ wip: state[ns]?.wip || false, err: state[ns]?.err || null });
+const getStats        = (state, ns, fn) => id => state[ns]?.[fn]?.byId[id] || 0;
+const getTree         = (state, ns, fn) => state[ns]?.[fn]?.tree || [];
+const isReady         = (state, ns) => !!state[ns]?.isReady;
+const getWipErr       = (state, ns) => ({ wip: state[ns]?.wip || false, err: state[ns]?.err || null });
+const citiesByCountry = (state, ns) => id => state[ns]?.[FN_LOCATIONS]?.citiesByCountry[id] || [];
```
Comment (bbfsdev) [inline src/redux/modules/filtersAside.js:185]: Where do you use this?

File: src/sagas/lists.js (hunk 1 above)
Comment (davgur) [reply src/sagas/lists.js:4]: I rename it from lesson to section, it can be one of sections (lesson, events...)

File: src/sagas/lists.js (hunk 2 above)
Comment (davgur) [reply src/sagas/lists.js:17]: its not redux namespace its page namespace (section), maybe better separate saga function instead add _ifs_

File: src/sagas/lists.js (hunk 3 above)
Comment (davgur) [reply src/sagas/lists.js:51]: yes, not need, all units are with views

File: src/sagas/lists.js (hunk 5 above)
Comment (davgur) [reply src/sagas/lists.js:66]: backend response was change so no _ content_units_ more

File: src/sagas/filtersAside.js (hunk 6 above)
Comment (davgur) [reply src/sagas/filtersAside.js:21]: what kind of filters needs counters. 
You mean add comment for describe API params?

File: src/sagas/filtersAside.js (hunk 7 above)
Comment (davgur) [reply src/sagas/filtersAside.js:67]: was not think that its possible, its same number as responses that was inserted
add validation?

File: src/redux/modules/filtersAside.js (hunk 8 above)
Comment (davgur) [reply src/redux/modules/filtersAside.js:185]: blin, on the next PR

**Review by bbfsdev — APPROVED** [review]

**Pull request merged**